core/trino-main/src/main/java/io/trino/sql/planner/EqualityInference.java

```diff
@@ -298,12 +298,9 @@ private Expression rewrite(Expression expression, Predicate<Symbol> symbolScope,
     /**
      * Returns the most preferrable expression to be used as the canonical expression
      */
-    private static Expression getCanonical(Collection<Expression> expressions)
+    private static Expression getCanonical(Stream<Expression> expressions)
```

Comment:
`Stream` isn't perfect as an input argument

Collection, Iterable, List are all better.
a collector would be better.

Since this is private helper, it's OK, but not in an api

Reply:
I understand this is not ideal, but we have calls like this:
```
getCanonical(equivalences.stream().filter(e -> isScoped(e, symbolScope)))
```
If we want to avoid collection, I see no other way.